q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,322 @@
+        (or the missing) value every instance which in one or many other columns
+        have a null or missing value.
+                    'validation:
+                      force_to_blank_if:
+                        is null:
+                        - variable'
+    normalization : dict
+        Apply the rules of normalizing the variables values, by replacing
+        the values outside the range by a given replacement value.
+                    'normalization:
+                        gated_value:
+                            - Out of bounds
+                        maximum:
+                            - int
+                            - float
+                        minimum:
+                            - int
+                            - float'
+    missing : str
+        Apply the rules of checking the allowed missing values, by replacing
+        the not-expected values and numpy's nan (empty) by the passed
+        missing value term.
+                    'missing:
+                        - not applicable
+                        - not collected
+                        - not provided
+                        - restricted access'
+    format : str
+        Apply the rules of checking that the passed column in the correct format.
+        [THIS COMMAND WOULD EITHER INFER THE DTYPE (as of now) OR USE THE
+        RESULT OF AN INITIAL DTYPE GETTER FUNCTION (see _dtypes.py)]
+                    'format:
+                      - bool
+                      - float
+                      - int
+                      - str'
+    expected : list
+        Apply the rules of allowing only expected values.
+        These do allow the missing and blanks.
+                    'expected:
+                        - str
+                        - list'
+    ontology : str
+        Name of the ontology to lookup within for control vocabulary, e.g.
+                    'ontology: Gazetteer'
+    blank : str
+        Apply the rules of checking the allowed blank values, by replacing
+        the not-expected values and numpy's nan (empty) by the passed
+        blank value term, e.g.
+                    'blank:
+                        - not applicable
+                        - not collected
+                        - not provided
+                        - restricted access'
+
+    Raises
+    ------
+    ValueError
+        # If the remote argument is not of ``bool`` or ``str`` type.
+        # If none of the samples in the ordination matrix are in the metadata.
+        # If the data is one-dimensional.
+    KeyError
+        # If there's samples in the ordination matrix but not in the metadata.
+    """
+    def __init__(self, variable_rules_fp: str):
+        """
+        Initialize the class instance for one set of rules beloning
+        to one single metadata variable with it possible attribute, as
+        a rate of one attribute per possible rule, plus the current
```

Comment:
as a rate of one attribute per possible rule?

Reply:
Good catch again - I meant "at a rate", but that remains too cryptic. Here's the new explanation:
```
        Initialize the class instance for the set of rules associated
        with a single metadata variable. The instantiated class will be
        given a `variable` attribute corresponding to the variable name
        as string, as well as a `rules` attribute collecting the actual
        rules in a dictionary. This dictionary has a hard-coded data 
        structure that has common default values for all variables and 
        that is then updated based on the parsed rules. 

```